relays/utils/src/lib.rs

```diff
@@ -271,3 +271,28 @@ where
 		},
 	}
 }
+
+/// An extension over rust's `Option` that adds an extra possibility: `Noop`.
```

Comment:
I would rather made it less generic - like something you have suggested [originally](https://github.com/paritytech/parity-bridges-common/pull/1419#discussion_r933856145) (or even `enum AvailableHeader<T> { Available(T), Missing, Unavailable }`, where `Missing` means `None`). It will (imo) add some context to the code + we may add more useful documentation to each option.

But please don't consider my opinion as the requirement - I'm fine with either solution, so please make your own decision here :)